packages/@aws-cdk/aws-apigateway/lib/rate-limited-api-key.ts

```diff
@@ -0,0 +1,132 @@
+import { Construct, IResource as IResourceBase, Resource } from '@aws-cdk/core';
+import { ApiKey } from './api-key';
+import { ResourceOptions } from "./resource";
+import { RestApi } from './restapi';
+import { QuotaSettings, ThrottleSettings, UsagePlan, UsagePlanPerApiStage } from './usage-plan';
+
+/**
+ * API keys are alphanumeric string values that you distribute to
+ * app developer customers to grant access to your API
+ */
+export interface IRateLimitedApiKey extends IResourceBase {
+  /**
+   * The API key ID.
+   * @attribute
+   */
+  readonly keyId: string;
+}
+
+/**
+ * Rate limiting settings for the api key.
+ */
+export interface RateLimitingSettings {
+  /**
+   * API Stages to be associated with the RateLimitedApiKey.
+   * @default none
+   */
+  readonly apiStages?: UsagePlanPerApiStage[];
+
+  /**
+   * Number of requests clients can make in a given time period.
+   * @default none
+   */
+  readonly quota?: QuotaSettings;
+
+  /**
+   * Overall throttle settings for the API.
+   * @default none
+   */
+  readonly throttle?: ThrottleSettings;
+}
+
+/**
+ * RateLimitedApiKey properties.
+ */
+export interface RateLimitedApiKeyProps extends ResourceOptions {
```

Comment:
If you extend `ApiKeyProps` instead, most of these properties don't need to be duplicated here.

Reply:
Good Idea! Done!